pkg/resource/deploy/step.go

```diff
@@ -892,8 +893,33 @@ func (s *ImportStep) Apply(preview bool) (resource.Status, StepCompleteFunc, err
 	s.diffs, s.detailedDiff = diff.ChangedKeys, diff.DetailedDiff
 
 	if diff.Changes != plugin.DiffNone {
-		const message = "inputs to import do not match the existing resource"
-
+		// Build an error message that says which keys have changed. First, pluck out the unique ones
+		// since nested structures may have multiple diffs in the array, and ensure a stable sort ordering.
+		var uniqueKeys []string
+		seenKeys := make(map[string]bool)
+		for _, key := range diff.ChangedKeys {
```

Comment:
If the detailed diff is available, I'd have this code prefer it over `ChangedKeys`. I am pretty sure that `ChangedKeys` is properly populated these days, but I'm not entirely confident.